### PR TITLE
Reduce test leftovers on disk

### PR DESCRIPTION
**Pull request by dothebart**
- don't dump agency by default on all errors
- don't write junit xml reports by default

### CODE REVIEW

**Comment by dothebart** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-quicky-linux/

**Comment by dothebart** [comment]
tests green except for recovery

File: js/client/modules/@arangodb/testing.js [hunk 1]
```diff
@@ -121,6 +122,7 @@ let optionsDocumentation = [
 ];
 
 const optionsDefaults = {
+  'dumpAgencyOnError': false,
```
Comment (jsteemann) [inline js/client/modules/@arangodb/testing.js]:
```suggestion
  'dumpAgencyOnError': true,
```

Comment (jsteemann) [reply js/client/modules/@arangodb/testing.js]: Let's flip the default here, as MaxN needs the dump.
Everyone can still turn off the dumps locally if required.

**Pull request merged**